docs/sources/mimir/query/query-metric-labels.md

```diff
@@ -0,0 +1,107 @@
+---
+description: Learn how to query metric labels from Grafana Mimir.
+title: Query metric labels
+menuTitle: Query metric labels
+weight: 2
+keywords:
+  - labels
+---
+
+# Query metric labels
+
+Grafana Mimir supports multiple [HTTP API]({{< relref "../references/http-api/index.md" >}}) endpoints to query metric label names and values.
+In this document we explain the main differences and trade-off between these API endpoints.
+
+## Querying label names
+
+The following API endpoints can be used to find the list of label names of all (or a subset of) series stored in Mimir:
+
+- **[Get label names]({{< relref "../references/http-api/index.md#get-label-names" >}})**<br />
+  `GET,POST <prometheus-http-prefix>/api/v1/labels`
+- **[Label names cardinality]({{< relref "../references/http-api/index.md#label-names-cardinality" >}})**<br />
+  `GET,POST <prometheus-http-prefix>/api/v1/cardinality/label_names`
+
+> **Note**: `<prometheus-http-prefix>` is the Prometheus HTTP prefix as documented in the [HTTP API]({{< relref "../references/http-api/index.md#path-prefixes" >}}) reference.
+> The default prefix is `/prometheus`.
+
+### Recommendations and performance trade-off
+
+- Prefer "get label names" over "label names cardinality" API, unless you need to know the number of unique label values for each label name.
+- The "get label names" API `start` and `end` parameters are optional. When omitted, Mimir will query label names for the whole retention period or up to the configured `-store.max-labels-query-length`. We recommend to always specify `start` and `end` parameters, and preferably set to the shortest period which is feasible for your use case.
+- For both API endpoints, executing a request specifying the series selector is computationally more expensive than the same request without the series selector, but the results set can be significantly smaller when the series selector effectively reduce the series set.
```

Comment:
```suggestion
- For both API endpoints, executing a request specifying the series selector is computationally more expensive than the same request without the series selector. However, the results set might be significantly smaller when the series selector effectively reduces the series set.
```
Can we remove "effectively" here? Do you mean "successfully"?

Reply:
We can remove it.